stl/inc/memory

```diff
@@ -3349,6 +3378,27 @@ public:
         return false;
     }
 
+    void wait(weak_ptr<_Ty> _Old, memory_order = memory_order::seq_cst) const noexcept {
+        for (;;) {
+            auto _Rep   = this->_Lock_and_load();
+            bool _Equal = this->_Ptr.load(memory_order::relaxed) == _Old._Ptr;
+            this->_Store_and_unlock(_Rep);
+            if (!_Equal) {
+                break;
+            }
+            __std_atomic_wait_direct(&this->_Ptr, &_Old._Ptr, sizeof(_Old._Ptr), _Atomic_wait_no_timeout);
+        }
+    }

```

Comment:
This is identical except for the type of `_Old`, between `atomic<shared_ptr>` and `atomic<weak_ptr>`. Can/should this be centralized in `_Atomic_ptr_base`, taking advantage of the fact that `shared_ptr` and `weak_ptr` are both `_Ptr_base`?

Reply:
It was pre-existing for other methods, still can be done here

Reply:
Centralized